packages/e2e-tests/scripts/save-examples.ts

```diff
@@ -215,23 +228,29 @@ async function makeReplayPublic(apiKey: string, recordingId: string) {
   });
 }
 
-async function waitUntilMessage(page: Page, message: string, timeout: number = 30_000) {
-  return await new Promise((resolve, reject) => {
-    let timer = setTimeout(reject, timeout);
+async function waitUntilMessage(
+  page: Page,
+  message: string,
+  timeout: number = 30_000
+): Promise<void> {
+  const gotMatch = await new Promise<boolean>(resolve => {
+    let timer = setTimeout(() => resolve(false), timeout);
```

Comment:
I'd like to better understand what motivated this change in behavior because it's not completely obvious to me. Before, if the operation timed out, it would thrown– and no it doesn't.

Seems like that's a significant change that might cause false positives in places like

https://github.com/replayio/devtools/blob/798464e945a7dd0e0139ea5e6e067a84747ee9fa/test/src/getExample.js#L32-L37

https://github.com/replayio/devtools/blob/798464e945a7dd0e0139ea5e6e067a84747ee9fa/test/src/getExample.js#L65-L70

Reply:
It does still throw, it's just a few lines farther down: https://github.com/replayio/devtools/blob/798464e945a7dd0e0139ea5e6e067a84747ee9fa/packages/e2e-tests/scripts/save-examples.ts#L252

The main motivation is that before this was essentially `throw undefined;` which made it really really hard for me to figure out what was throwing when this failed.

Reply:
Ahh, fair enough! I overlooked that.